src/nfts/NftsInfoCarousel.tsx

```diff
@@ -0,0 +1,350 @@
+          }}
+          onError={() => {
+            Logger.error(
+              TAG,
+              `Error loading Nft thumbnail image for Nft contractAddress: ${nft.contractAddress} tokenId: ${nft.tokenId}`
+            )
+          }}
+          resizeMode={FastImage.resizeMode.cover}
+          onLoadEnd={() => {
+            setLoading(false)
+          }}
+        >
+          {loading && <ThumbnailImagePlaceholder />}
+        </FastImage>
+      )}
+    </Touchable>
+  )
+}
+
+export default function NftsInfoCarousel({ route }: Props) {
+  const { nfts } = route.params
+  const [activeNft, setActiveNft] = useState(nfts[0] ?? null)
+  const [isLoading, setIsLoading] = useState(true)
+  const [scaledHeight, setScaledHeight] = useState(360)
+  const { t } = useTranslation()
+
+  function pressExplorerLink() {
+    navigate(Screens.WebViewScreen, {
+      uri: `${networkConfig.celoExplorerBaseTokenUrl}${activeNft.contractAddress}/instance/${activeNft.tokenId}/metadata`,
+    })
+  }
+
+  const MainImagePlaceholder = () => {
+    return (
+      <SkeletonPlaceholder
+        borderRadius={8}
+        backgroundColor={colors.gray2}
+        highlightColor={colors.white}
+      >
+        <View
+          style={{
+            height: scaledHeight,
+            width: variables.width,
+            zIndex: -1,
+          }}
+        />
+      </SkeletonPlaceholder>
+    )
+  }
+
+  const NftImageCarousel = () => {
+    const handleOnPress = (nft: Nft) => () => {
+      setActiveNft(nft)
+    }
+    return (
+      <View style={styles.nftImageCarouselContainer}>
+        <ScrollView
+          showsHorizontalScrollIndicator={false}
+          horizontal={true}
+          contentContainerStyle={styles.carouselScrollViewContentContainer}
+          style={styles.nftImageCarousel}
+          testID="NftsInfoCarousel/NftImageCarousel"
+        >
+          {nfts.map((nft) => {
+            return (
+              <View
+                key={`${nft.contractAddress}-${nft.tokenId}`}
+                style={styles.nftThumbnailSharedContainer}
+              >
+                <NftThumbnail
+                  onPress={handleOnPress(nft)}
+                  // Use contractAddress and tokenId for a unique key
+                  isActive={
+                    `${activeNft.contractAddress}-${activeNft.tokenId}` ===
+                    `${nft.contractAddress}-${nft.tokenId}`
+                  }
+                  // If there's no nft metadata, show a red info icon instead
+                  error={!nft.metadata}
+                  nft={nft}
+                />
+              </View>
+            )
+          })}
+        </ScrollView>
+      </View>
+    )
+  }
+
+  // Display error screen no Nfts provided
+  if (nfts.length === 0) {
+    return <NftsLoadErrorScreen />
+  }
+
+  return (
+    <SafeAreaView edges={['top']} style={styles.safeAreaView} testID="NftsInfoCarousel">
```

Comment:
any reason why we don't want safe area on the other edges too? i'm always confused by this - when is it nicer for content to be displayed overlapping a bottom bar on iOS for example?

Reply:
We have something similar with `src/home/WalletHome.tsx`. I tried without and it's a bit strange that all the content is scrolling except a sliver at the bottom of the screen.